storm-core/src/jvm/backtype/storm/utils/RateTracker.java

```diff
@@ -72,7 +80,7 @@ public RateTracker(int validTimeWindowInMils, int numOfSlides, boolean simulate
      * @param count number of arrivals
      */
     public void notify(long count) {
-        _histograms[_histograms.length-1]+=count;
+        _histograms[_numOfSlides - 1] += count;
```

Comment:
I don't think this is thread safe.  I'm not sure that is a big deal for this code, as I am not aware of any spouts that emit from multiple threads ignoring the call to nextTuple.  But we should file a JIRA at least to fix it because the disruptor code calls it from multiple threads, and the counts will get off.